src/main/java/ch/njol/skript/expressions/ExprVehicle.java

```diff
@@ -38,102 +29,108 @@
-				return ((VehicleExitEvent) e).getVehicle();
+			if (getTime() >= 0 && event instanceof VehicleExitEvent && entity.equals(((VehicleExitEvent) event).getExited()) && !Delay.isDelayed(event)) {
+				return ((VehicleExitEvent) event).getVehicle();
 			}
 			if (hasMountEvents) {
-				if (getTime() >= 0 && e instanceof EntityMountEvent && entity.equals(((EntityMountEvent) e).getEntity()) && !Delay.isDelayed(e)) {
-					return ((EntityMountEvent) e).getMount();
+				if (getTime() >= 0 && event instanceof EntityMountEvent && entity.equals(((EntityMountEvent) event).getEntity()) && !Delay.isDelayed(event)) {
+					return ((EntityMountEvent) event).getMount();
 				}
-				if (getTime() >= 0 && e instanceof EntityDismountEvent && entity.equals(((EntityDismountEvent) e).getEntity()) && !Delay.isDelayed(e)) {
-					return ((EntityDismountEvent) e).getDismounted();
+				if (getTime() >= 0 && event instanceof EntityDismountEvent && entity.equals(((EntityDismountEvent) event).getEntity()) && !Delay.isDelayed(event)) {
+					return ((EntityDismountEvent) event).getDismounted();
 				}
 			}
 			return entity.getVehicle();
 		});
 	}
-	
+
 	@Override
 	@Nullable
-	public Entity convert(final Entity e) {
+	public Entity convert(Entity entity) {
 		assert false;
-		return e.getVehicle();
-	}
-	
-	@Override
-	public Class<? extends Entity> getReturnType() {
-		return Entity.class;
-	}
-	
-	@Override
-	protected String getPropertyName() {
-		return "vehicle";
+		return entity.getVehicle();
 	}
-	
+
 	@Override
 	@Nullable
-	public Class<?>[] acceptChange(final ChangeMode mode) {
-		if (mode == ChangeMode.SET) {
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.SET) 
 			return new Class[] {Entity.class, EntityData.class};
-		}
 		return super.acceptChange(mode);
 	}
-	
+
 	@Override
-	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
 		if (mode == ChangeMode.SET) {
 			assert delta != null;
-			final Entity[] ps = getExpr().getArray(e);
-			if (ps.length == 0)
+			Entity[] passengers = getExpr().getArray(event);
+			if (passengers.length == 0)
 				return;
-			final Object o = delta[0];
-			if (o instanceof Entity) {
-				((Entity) o).eject();
-				final Entity p = CollectionUtils.getRandom(ps);
-				assert p != null;
-				p.leaveVehicle();
-				((Entity) o).setPassenger(p);
-			} else if (o instanceof EntityData) {
-				for (final Entity p : ps) {
-					final Entity v = ((EntityData<?>) o).spawn(p.getLocation());
-					if (v == null)
+			Object object = delta[0];
+			if (object instanceof Entity) {
+				((Entity) object).eject();
+				Entity passenger = CollectionUtils.getRandom(passengers);
+				assert passenger != null;
+				passenger.leaveVehicle();
+				((Entity) object).setPassenger(passenger);
```

Comment:
Should be using addPassenger instead of setPassenger, no? (goes for all uses of setPassenger)